From 498ba97ba1897cd4a39f92c1732e7127047a23b3 Mon Sep 17 00:00:00 2001
From: Yuval Caduri <cyuval@marvell.com>
Date: Sun, 4 Feb 2018 10:53:05 +0200
Subject: [PATCH 2229/2241] marvell: dts: Add high-memory cma shared-pool

Add reserved memory-region in AP1 DDR (upper 8GB).
This region may be used for coherent dma allocations
in drivers/devices that use CP4, which is attached to AP1.

Change-Id: Icf20a1540b0b0a7952cb27dac8f15043e203d73b
Signed-off-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/49967
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-8162-db.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8162-db.dts b/arch/arm64/boot/dts/marvell/armada-8162-db.dts
index 935f453..7b7ad06 100644
--- a/arch/arm64/boot/dts/marvell/armada-8162-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8162-db.dts
@@ -55,6 +55,18 @@
 		stdout-path = "serial0:115200n8";
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* global autoconfigured region for contiguous allocations */
+		ap1_mem: ap1-area@4230000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x42 0x30000000 0 0x10000000>;
+		};
+	};
 	aliases {
 		ethernet0 = &cp0_emac0;
 		ethernet1 = &cp0_emac2;
@@ -142,6 +154,7 @@
 &cp4_ppv22 {
 	l4_chksum_jumbo_port = <0>;
 	pinctrl-names = "default";
+	memory-region = <&ap1_mem>;
 };
 
 &cp4_eth0 {
-- 
2.7.4

